### ISSUE TITLE
[checkbox] Use pseudo checkbox in lieu of hidden input

### ISSUE THREAD

**Issue by jaredpalmer** [reach/reach-ui#491]
### Current Behavior

Checkboxes are great and are a11y-friendly!

### Desired Behavior

Often times with checkbox groups (and even radio groups), questions have an "Other" option. When chosen, a text input becomes visible and focused in the UI. The user can then enter a custom value. If they then check another option, the Other option's `<input>` should remain with the value they typed. 

### Suggested Solution

This behavior is surprisingly tricky to get right because ultimately the value of the radio (or element in the array for a checkbox group) and the input need to either sync correctly or be separated and reconciled later (during form submission).

### Who does this impact? Who is this for?

Pretty common survey behavior.

### Describe alternatives you've considered

Even if we do not support this behavior directly, perhaps there is a way we can allow for it with some kind of pseudo-checkbox/radio (so it is safely cloned and gets the correct keyboard interactions.

**Comment by chancestrickland** [comment]
Thanks for the writeup! Just to be clear, is the issue here related to focusing the "other" text field when using a `CustomCheckbox` (we're using `requestAnimationFrame` to hack around the browser's native behavior a bit)?

**Comment by jaredpalmer** [comment]
yes, that's what's wonky right now

**Comment by chancestrickland** [comment]
I hear you, and while we may (read: will likely) revisit our implementation to avoid the focus battle, it might be good to know that the pattern you've described, while common, could actually be a WCAG violation. You probably don't want to change the focus when the new input appears, rather it should be the next element in the tab order https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html

We are using containers in @reach/radio instead of wrapping native inputs, so it probably makes sense to do it here too to make focus management a little less painful.
